### PR TITLE
SCMOD-9861: Ignore unsupported mapping changes

### PR DESCRIPTION
**Pull request by kusumaghoshdastidar**
Jira: https://portal.digitalsafe.net/browse/SCMOD-9861

### CODE REVIEW

File: release-notes-1.1.0.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ ${version-number}
 
 #### New Features
 - **SCMOD-9780**: Updated images to use Java 11
+- **SCMOD-9861**: Updated to ignore unsupported mapping changes
```
Comment (dermot-hardy) [inline release-notes-1.1.0.md:7]: I think I'd count this as a breaking change.  Let's move to 2.0.0-SNAPSHOT for this.

File: elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java [hunk 2]
```diff
@@ -215,34 +217,75 @@ private static boolean isMappingChangeSafe(final Map<String, Object> templateMap
...
+                            key, value.rightValue(), value.leftValue());
+                allowedFieldDifferences.remove(getFieldName(key));
+            });
             return false;
         }
     }
 
+    private static String getFieldName(final String key)
+    {
+        return key.split(Pattern.quote("/"))[1];
```
[the remark this answers is not in the document]
Comment (kusumaghoshdastidar) [reply elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java:231]: The key names are flattened to: /LANGUAGE_CODES/CODE/type. Once split the second token has the field name 'LANGUAGE_CODES'.

[the remark this answers is not in the document]
Comment (dermot-hardy) [reply elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java:231]: So for the sake of argument if `/ENTITIES/MATCHES` changed from nested to object does that means that a supported change to `/ENTITIES/GRAMMAR_ID` wouldn't be applied?

[the remark this answers is not in the document]
Comment (kusumaghoshdastidar) [reply elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java:231]: Yes

[the remark this answers is not in the document]
Comment (dermot-hardy) [reply elastic-mapping-updater/src/main/java/com/github/cafdataprocessing/elastic/tools/ElasticMappingUpdater.java:231]: Ok I've raised [SCMOD-10180](https://portal.digitalsafe.net/browse/SCMOD-10180) for that.

**Pull request merged**